pygbm/splitting.py

```diff
@@ -234,19 +253,19 @@ def find_node_split_subtraction(context, sample_indices, parent_histograms,
     # compute the gradients: they must be the same across all features
     # anyway, we have tests ensuring this. Maybe a more robust way would
     # be to compute an average but it's probably not worth it.
-    sum_gradients = (parent_histograms[0]['sum_gradients'].sum() -
-                     sibling_histograms[0]['sum_gradients'].sum())
+    context.sum_gradients = (parent_histograms[0]['sum_gradients'].sum() -
+                             sibling_histograms[0]['sum_gradients'].sum())
 
     if context.constant_hessian:
         n_samples = sample_indices.shape[0]
-        sum_hessians = context.constant_hessian_value * float32(n_samples)
+        context.sum_hessians = \
+            context.constant_hessian_value * float32(n_samples)
     else:
-        sum_hessians = (parent_histograms[0]['sum_hessians'].sum() -
-                        sibling_histograms[0]['sum_hessians'].sum())
+        context.sum_hessians = (parent_histograms[0]['sum_hessians'].sum() -
+                                sibling_histograms[0]['sum_hessians'].sum())
 
-    return _parallel_find_split_subtraction(
-        context, parent_histograms, sibling_histograms,
-        sum_gradients, sum_hessians)
+    return _parallel_find_split_subtraction(context, parent_histograms,
+                                            sibling_histograms)
```

Comment:
Similar remark as above w.r.t. collapsing function and reusing the docstring. Don't forget to add the `parallel=True` flag to `find_node_split_subtraction` when you do this.